Panic in Volume.createIfNotExist

I'm running build from master, getting the next error:

```
time="2015-01-21T19:30:34+03:00" level="warning" msg="WARNING: Udev sync is not supported. This will lead to unexpected behavior, data loss and errors"
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x20 pc=0x73c61c]

goroutine 9 [running]:
github.com/docker/docker/volumes.(*Volume).createIfNotExist(0xc2085071d0, 0x0, 0x0)
	/go/src/github.com/docker/docker/volumes/volume.go:91 +0xbc
github.com/docker/docker/volumes.(*Volume).initialize(0xc2085071d0, 0x0, 0x0)
	/go/src/github.com/docker/docker/volumes/volume.go:111 +0x9a
github.com/docker/docker/volumes.(*Repository).restore(0xc20850a1e0, 0x0, 0x0)
	/go/src/github.com/docker/docker/volumes/repository.go:100 +0x402
github.com/docker/docker/volumes.NewRepository(0xc208203980, 0x17, 0x7fc4d2e77c40, 0xc2081a68f0, 0x17, 0x0, 0x0)
	/go/src/github.com/docker/docker/volumes/repository.go:39 +0x1fa
github.com/docker/docker/daemon.NewDaemonFromDirectory(0x12d7c40, 0xc208078a50, 0x0, 0x0, 0x0)
	/go/src/github.com/docker/docker/daemon/daemon.go:892 +0x1114
github.com/docker/docker/daemon.NewDaemon(0x12d7c40, 0xc208078a50, 0x0, 0x0, 0x0)
	/go/src/github.com/docker/docker/daemon/daemon.go:779 +0x44
main.func·001()
	/go/src/github.com/docker/docker/docker/daemon.go:55 +0x46
created by main.mainDaemon
	/go/src/github.com/docker/docker/docker/daemon.go:78 +0x35e
```

Problematic snippet of code:

```go
	if stat, err := os.Stat(v.Path); err != nil && os.IsNotExist(err) {
		if stat.IsDir() {
			os.MkdirAll(v.Path, 0755)
		}
	// ...
	}
```

`err` is actually `stat error: stat : no such file or directory`.

--- a/daemon/start.go
+++ b/daemon/start.go
@@ -53,6 +53,8 @@ func (daemon *Daemon) setHostConfig(container *Container, hostConfig *runconfig.
 	if err := parseSecurityOpt(container, hostConfig); err != nil {
 		return err
 	}
+
+	// FIXME: this should be handled by the volume subsystem
 	// Validate the HostConfig binds. Make sure that:
 	// the source exists
 	for _, bind := range hostConfig.Binds {

--- a/volumes/volume.go
+++ b/volumes/volume.go
@@ -86,30 +86,14 @@ func (v *Volume) AddContainer(containerId string) {
 	v.lock.Unlock()
 }
 
-func (v *Volume) createIfNotExist() error {
-	if stat, err := os.Stat(v.Path); err != nil && os.IsNotExist(err) {
-		if stat.IsDir() {
-			os.MkdirAll(v.Path, 0755)
-		}
-
-		if err := os.MkdirAll(filepath.Dir(v.Path), 0755); err != nil {
-			return err
-		}
-		f, err := os.OpenFile(v.Path, os.O_CREATE, 0755)
-		if err != nil {
-			return err
-		}
-		f.Close()
-	}
-	return nil
-}
-
 func (v *Volume) initialize() error {
 	v.lock.Lock()
 	defer v.lock.Unlock()
 
-	if err := v.createIfNotExist(); err != nil {
-		return err
+	if _, err := os.Stat(v.Path); err != nil && os.IsNotExist(err) {
+		if err := os.MkdirAll(v.Path, 0755); err != nil {
+			return err
+		}
 	}
 
 	if err := os.MkdirAll(v.configPath, 0755); err != nil {
@@ -133,6 +117,7 @@ func (v *Volume) ToDisk() error {
 	defer v.lock.Unlock()
 	return v.toDisk()
 }
+
 func (v *Volume) toDisk() error {
 	data, err := json.Marshal(v)
 	if err != nil {
@@ -146,6 +131,7 @@ func (v *Volume) toDisk() error {
 
 	return ioutil.WriteFile(pth, data, 0666)
 }
+
 func (v *Volume) FromDisk() error {
 	v.lock.Lock()
 	defer v.lock.Unlock()

